Installation issue: r@4.5.0

### Steps to reproduce the issue

```console
$ spack spec -I r
 -   r@4.5.0~X~memory_profiling~rmath build_system=autotools patches:=abc572d arch=linux-centos7-ivybridge %c,cxx,fortran=gcc@9.5.0                                                                                     
[+]      ^bzip2@1.0.8~debug~pic+shared build_system=generic arch=linux-centos7-ivybridge %c=gcc@9.5.0                                                                                                                   
[+]          ^diffutils@3.10 build_system=autotools arch=linux-centos7-ivybridge %c=gcc@9.5.0                                                                                                                           
[+]      ^compiler-wrapper@1.0 build_system=generic arch=linux-centos7-ivybridge                                                                                                                                        
[+]      ^curl@8.11.1~gssapi~ldap+libidn2~librtmp~libssh+libssh2+nghttp2 build_system=autotools libs:=shared,static tls:=mbedtls arch=linux-centos7-ivybridge %c,cxx=gcc@9.5.0                                          
[+]          ^libidn2@2.3.7 build_system=autotools arch=linux-centos7-ivybridge %c=gcc@9.5.0                                                                                                                            
[+]              ^libunistring@1.2 build_system=autotools arch=linux-centos7-ivybridge %c=gcc@9.5.0                                                                                                                     
[+]          ^libssh2@1.11.1+shared build_system=autotools crypto=mbedtls arch=linux-centos7-ivybridge %c=gcc@9.5.0                                                                                                     
[+]          ^mbedtls@2.28.9+pic build_system=makefile build_type=Release libs:=shared,static arch=linux-centos7-ivybridge %c=gcc@9.5.0                                                                                 
[+]          ^nghttp2@1.59.0 build_system=autotools arch=linux-centos7-ivybridge %c,cxx=gcc@9.5.0                                                                                                                       
[+]          ^pkgconf@2.3.0 build_system=autotools arch=linux-centos7-ivybridge %c=gcc@9.5.0                                                                                                                            
[e]      ^gcc@9.5.0~binutils+bootstrap~graphite~nvptx~piclibs~profiled~strip build_system=autotools build_type=RelWithDebInfo languages:='c,c++,fortran' patches:=0d13622,d4919d6 arch=linux-centos7-ivybridge                     
[+]      ^gcc-runtime@9.5.0 build_system=generic arch=linux-centos7-ivybridge                                                                                                                                                          
[+]      ^gettext@0.23.1+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools arch=linux-centos7-ivybridge %c,cxx=gcc@9.5.0                                                                   
[+]          ^libiconv@1.18 build_system=autotools libs:=shared,static arch=linux-centos7-ivybridge %c=gcc@9.5.0                                                                                                         
[+]          ^libxml2@2.13.5~http+pic~python+shared build_system=autotools arch=linux-centos7-ivybridge %c=gcc@9.5.0                                                                                                    
[+]          ^tar@1.35 build_system=autotools zip=pigz arch=linux-centos7-ivybridge %c=gcc@9.5.0                                                                                                                        
[+]              ^pigz@2.8 build_system=makefile arch=linux-centos7-ivybridge %c=gcc@9.5.0                                                                                                                              
[e]      ^glibc@2.17 build_system=autotools patches:=be65fec,e179c43 arch=linux-centos7-ivybridge                                                                                                                        
[+]      ^gmake@4.4.1~guile build_system=generic arch=linux-centos7-ivybridge %c=gcc@9.5.0                                                                                                                               
[+]      ^icu4c@74.2 build_system=autotools cxxstd=11 arch=linux-centos7-ivybridge %c,cxx=gcc@9.5.0                                                                                                                        
[+]          ^autoconf@2.72 build_system=autotools arch=linux-centos7-ivybridge                                                                                                                                              
[+]              ^m4@1.4.19+sigsegv build_system=autotools patches:=9dc5fbd,bfdffa7 arch=linux-centos7-ivybridge %c,cxx=gcc@9.5.0                                                                                               
[+]                  ^libsigsegv@2.14 build_system=autotools arch=linux-centos7-ivybridge %c=gcc@9.5.0                                                                                                                             
[+]          ^automake@1.16.5 build_system=autotools arch=linux-centos7-ivybridge %c=gcc@9.5.0                                                                                                                          
[+]          ^libtool@2.4.7 build_system=autotools arch=linux-centos7-ivybridge %c=gcc@9.5.0                                                                                                                             
[+]              ^findutils@4.10.0 build_system=autotools patches:=440b954 arch=linux-centos7-ivybridge %c=gcc@9.5.0                                                                                                       
[+]          ^python@3.13.2+bz2+ctypes+dbm~debug+libxml2+lzma~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic arch=linux-centos7-ivybridge %c,cxx=gcc@9.5.0       
[+]              ^expat@2.7.1+libbsd build_system=autotools arch=linux-centos7-ivybridge %c,cxx=gcc@9.5.0                                                                                                               
[+]                  ^libbsd@0.12.2 build_system=autotools arch=linux-centos7-ivybridge %c=gcc@9.5.0                                                                                                                    
[+]                      ^libmd@1.1.0 build_system=autotools arch=linux-centos7-ivybridge %c=gcc@9.5.0                                                                                                                          
[+]              ^gdbm@1.23 build_system=autotools arch=linux-centos7-ivybridge %c=gcc@9.5.0                                                                                                                                       
[+]              ^libffi@3.4.8 build_system=autotools arch=linux-centos7-ivybridge %c,cxx=gcc@9.5.0                                                                                                                                    
[+]              ^openssl@3.4.1~docs+shared build_system=generic certs=mozilla arch=linux-centos7-ivybridge %c,cxx=gcc@9.5.0                                                                                             
[+]                  ^ca-certificates-mozilla@2025-02-25 build_system=generic arch=linux-centos7-ivybridge                                                                                                               
[+]              ^sqlite@3.46.0+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=linux-centos7-ivybridge %c=gcc@9.5.0                                                                 
[+]              ^util-linux-uuid@2.41 build_system=autotools arch=linux-centos7-ivybridge %c=gcc@9.5.0                                                                                                                      
[+]      ^libtirpc@1.3.3 build_system=autotools arch=linux-centos7-ivybridge %c=gcc@9.5.0                                                                                                                               
[+]          ^krb5@1.21.3+shared build_system=autotools arch=linux-centos7-ivybridge %c,cxx=gcc@9.5.0                                                                                                                      
[+]              ^bison@3.8.2~color build_system=autotools arch=linux-centos7-ivybridge %c,cxx=gcc@9.5.0                                                                                                                 
[+]              ^libedit@3.1-20240808 build_system=autotools arch=linux-centos7-ivybridge %c=gcc@9.5.0                                                                                                                  
[+]      ^ncurses@6.5~symlinks+termlib abi=none build_system=autotools patches:=7a351bc arch=linux-centos7-ivybridge %c,cxx=gcc@9.5.0                                                                                      
[+]      ^openblas@0.3.29~bignuma~consistent_fpcsr+dynamic_dispatch+fortran+ilp64+locking+pic+shared build_system=makefile symbol_suffix=64_ threads=none arch=linux-centos7-ivybridge %c,cxx,fortran=gcc@9.5.0              
[+]      ^openjdk@21.0.3_9 build_system=generic certs=none arch=linux-centos7-ivybridge                                                                                                                                 
[+]      ^pcre2@10.44~jit+multibyte+pic build_system=autotools arch=linux-centos7-ivybridge %c=gcc@9.5.0                                                                                                                
[+]      ^readline@8.2 build_system=autotools patches:=1ea4349,24f587b,3d9885e,5911a5b,622ba38,6c8adf8,758e2ec,79572ee,a177edc,bbf97f1,c7b45ff,e0013d9,e065038 arch=linux-centos7-ivybridge %c=gcc@9.5.0                   
[+]      ^texinfo@7.1 build_system=autotools arch=linux-centos7-ivybridge %c,cxx=gcc@9.5.0                                                                                                                               
[+]          ^perl@5.40.0+cpanm+opcode+open+shared+threads build_system=generic arch=linux-centos7-ivybridge %c=gcc@9.5.0                                                                                               
[+]              ^berkeley-db@18.1.40+cxx~docs+stl build_system=autotools patches:=26090f4,b231fcc arch=linux-centos7-ivybridge %c,cxx=gcc@9.5.0                                                                             
[+]      ^which@2.21 build_system=autotools arch=linux-centos7-ivybridge %c=gcc@9.5.0                                                                                                                                   
[+]      ^xz@5.6.3~pic build_system=autotools libs:=shared,static arch=linux-centos7-ivybridge %c=gcc@9.5.0                                                                                                                
[+]      ^zlib-ng@2.2.4+compat+new_strategies+opt+pic+shared build_system=autotools arch=linux-centos7-ivybridge %c,cxx=gcc@9.5.0                                                                                            
[+]      ^zstd@1.5.7+programs build_system=makefile compression:=none libs:=shared,static arch=linux-centos7-ivybridge %c,cxx=gcc@9.5.0                                              
```


### Error message

<details><summary>Error message</summary>

`spack -k install` attempt (`-k` due to institution proxy):

```
==> No binary for r-4.5.0-7mlsop2kfhmm4kzslpvdzhr3sw7cwcmm found: installing from source
==> Installing r-4.5.0-7mlsop2kfhmm4kzslpvdzhr3sw7cwcmm [74/75]
[+] /gpfs/fs001/cbica/software/centos7/spack/opt/spack/linux-ivybridge/julia-1.11.5-lbxvmfigl7a7dkjcny3wrq3fdghhpsen
==> Using cached archive: /gpfs/fs001/cbica/software/centos7/spack/var/spack/cache/_source-cache/archive/3b/3b33ea113e0d1ddc9793874d5949cec2c7386f66e4abfb1cef9aec22846c3ce1.tar.gz
==> Applied patch /cbica/home/sysswadm/.spack/package_repos/fncqgg4/repos/spack_repo/builtin/packages/r/relocate-which.patch
==> r: Executing phase: 'autoreconf'
==> r: Executing phase: 'configure'
==> Error: ProcessError: Command exited with status 1:
    '/scratch/sysswadm/tmp.GUTWkqRIJN/spack-stage/spack-stage-r-4.5.0-7mlsop2kfhmm4kzslpvdzhr3sw7cwcmm/spack-src/configure' '--prefix=/gpfs/fs001/cbica/software/centos7/spack/opt/spack/linux-ivybridge/r-4.5.0-7mlsop2kfhmm4kzslpvdzhr3sw7cwcmm' '--with-internal-tzcode' '--libdir=/gpfs/fs001/cbica/software/centos7/spack/opt/spack/linux-ivybridge/r-4.5.0-7mlsop2kfhmm4kzslpvdzhr3sw7cwcmm/rlib' '--enable-R-shlib' '--enable-R-framework=no' '--without-recommended-packages' 'LDFLAGS=-Wl,-rpath,/gpfs/fs001/cbica/software/centos7/spack/opt/spack/linux-ivybridge/r-4.5.0-7mlsop2kfhmm4kzslpvdzhr3sw7cwcmm/rlib/R/lib' '--with-blas=-L/gpfs/fs001/cbica/software/centos7/spack/opt/spack/linux-ivybridge/openblas-0.3.29-ba5xf562sisskrqtrfixwevn6pi6jklc/lib -lopenblas64_' '--with-lapack=-L/gpfs/fs001/cbica/software/centos7/spack/opt/spack/linux-ivybridge/openblas-0.3.29-ba5xf562sisskrqtrfixwevn6pi6jklc/lib -lopenblas64_' 'ac_cv_path_PDFLATEX=' 'ac_cv_path_PDFTEX=' 'ac_cv_path_TEX=' 'ac_cv_path_TEXI2DVI=' '--with-libintl-prefix=/gpfs/fs001/cbica/software/centos7/spack/opt/spack/linux-ivybridge/gettext-0.23.1-mxyqfccn34ykqikw72frzbzvwd7o4pad' '--without-cairo' '--without-jpeglib' '--without-libpng' '--without-libtiff' '--without-tcltk' '--without-x'

1 error found in build log:
     427    checking whether csinh exists and is declared... yes
     428    checking whether ctanh exists and is declared... yes
     429    checking for working ctanh... yes
     430    checking whether 'struct tm' includes tm_zone... yes
     431    checking whether 'struct tm' includes tm_gmtoff... yes
     432    checking for dgemm_ in -L/gpfs/fs001/cbica/software/centos7/spack/opt/spack/linux-ivybridge/openblas-0.3.29-ba5xf562sisskrqtrfixwevn6pi6jklc/lib -lopenblas64_... no
  >> 433    configure: error: BLAS was specified but not available

See build log for details:
  /scratch/sysswadm/tmp.GUTWkqRIJN/spack-stage/spack-stage-r-4.5.0-7mlsop2kfhmm4kzslpvdzhr3sw7cwcmm/spack-build-out.txt

==> Error: r-4.5.0-7mlsop2kfhmm4kzslpvdzhr3sw7cwcmm: ProcessError: Command exited with status 1:
    '/scratch/sysswadm/tmp.GUTWkqRIJN/spack-stage/spack-stage-r-4.5.0-7mlsop2kfhmm4kzslpvdzhr3sw7cwcmm/spack-src/configure' '--prefix=/gpfs/fs001/cbica/software/centos7/spack/opt/spack/linux-ivybridge/r-4.5.0-7mlsop2kfhmm4kzslpvdzhr3sw7cwcmm' '--with-internal-tzcode' '--libdir=/gpfs/fs001/cbica/software/centos7/spack/opt/spack/linux-ivybridge/r-4.5.0-7mlsop2kfhmm4kzslpvdzhr3sw7cwcmm/rlib' '--enable-R-shlib' '--enable-R-framework=no' '--without-recommended-packages' 'LDFLAGS=-Wl,-rpath,/gpfs/fs001/cbica/software/centos7/spack/opt/spack/linux-ivybridge/r-4.5.0-7mlsop2kfhmm4kzslpvdzhr3sw7cwcmm/rlib/R/lib' '--with-blas=-L/gpfs/fs001/cbica/software/centos7/spack/opt/spack/linux-ivybridge/openblas-0.3.29-ba5xf562sisskrqtrfixwevn6pi6jklc/lib -lopenblas64_' '--with-lapack=-L/gpfs/fs001/cbica/software/centos7/spack/opt/spack/linux-ivybridge/openblas-0.3.29-ba5xf562sisskrqtrfixwevn6pi6jklc/lib -lopenblas64_' 'ac_cv_path_PDFLATEX=' 'ac_cv_path_PDFTEX=' 'ac_cv_path_TEX=' 'ac_cv_path_TEXI2DVI=' '--with-libintl-prefix=/gpfs/fs001/cbica/software/centos7/spack/opt/spack/linux-ivybridge/gettext-0.23.1-mxyqfccn34ykqikw72frzbzvwd7o4pad' '--without-cairo' '--without-jpeglib' '--without-libpng' '--without-libtiff' '--without-tcltk' '--without-x'
==> Error: r-4.5.0-7mlsop2kfhmm4kzslpvdzhr3sw7cwcmm: Package was not installed
==> Error: Installation request failed.  Refer to reported errors for failing package(s).
```
</details>


### Information on your system

* **Spack:** 1.0.0.dev0 (https://github.com/spack/spack/commit/c702f5b89eea1950c76472fbf8f0919ad12880a6)
* **Builtin repo:** https://github.com/spack/spack-packages/commit/050e78d90ffe47013ac84f8cf10b31307d9b76f3
* **Python:** 3.11.13
* **Platform:** linux-centos7-ivybridge


### Additional information

[spack-build-env.txt](https://github.com/user-attachments/files/21005072/spack-build-env.txt)
[spack-build-out.txt](https://github.com/user-attachments/files/21005071/spack-build-out.txt)

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

--- a/repos/spack_repo/builtin/packages/r/package.py
+++ b/repos/spack_repo/builtin/packages/r/package.py
@@ -83,7 +83,7 @@ class R(AutotoolsPackage):
     depends_on("fortran", type="build")
 
     depends_on("blas")
-    requires("^openblas +symbol_suffix=none", when="^openblas")
+    requires("^openblas symbol_suffix=none", when="^openblas")
     depends_on("lapack")
 
     depends_on("bzip2")

--- a/repos/spack_repo/builtin/packages/r/package.py
+++ b/repos/spack_repo/builtin/packages/r/package.py
@@ -83,6 +83,7 @@ class R(AutotoolsPackage):
     depends_on("fortran", type="build")
 
     depends_on("blas")
+    requires("^openblas +symbol_suffix=none", when="^openblas")
     depends_on("lapack")
 
     depends_on("bzip2")

